extensions/components/com_redcore/admin/helpers/translation.php

```diff
@@ -107,4 +107,61 @@ public static function getTranslationItemStatus($item, $columns)
 			return $translationStatus;
 		}
 	}
+
+	/**
+	 * Gets translation item id and returns it
+	 *
+	 * @param   int     $itemId            Item id
+	 * @param   string  $langCode          Language code
+	 * @param   string  $pk  			   Primary key name
+	 *
+	 * @return  int     Translations item id
+	 */
+	public static function getTranslationItemId($itemid, $langCode, $pk) 
+	{
+		$table = RedcoreHelpersTranslation::getTranslationTable();
+		
+		$ids = explode('###', $itemid);
+
+		$db = JFactory::getDbo();
+		$query = $db->getQuery(true)
+		->select('rctranslations_id')
+		->from($db->qn(RTranslationTable::getTranslationsTableName($table->table, '')))
+		->where('rctranslations_language=' . $db->q($langCode));
+		foreach ($pk as $key => $primaryKey)
+		{
+			$query->where($db->qn($primaryKey) . ' = ' . $db->q($ids[$key]));
+		}
+
+		$db->setQuery($query);
+
+		return $db->loadResult();	
+	}
+
+	/**
+	 * Checks if an array of data is empty
+	 *
+	 * @param   array    $data       Array of data to be checked
+	 * @param   array    $excludes   Array of keys to be excluded from validation
+	 *
+	 * @return  boolean 
+	 */
+	public static function validateEmptyTranslationData($data, $excludes = null)
+	{
+		//Remove excluded keys from array
+		foreach ($excludes as $exclude)
+		{
+			unset($data[$exclude]);
+		}
+
+		//Check if the rest of the keys in the array are empty
+		if (array_filter($data)) 
```

Comment:
there is a problem using this approach... one is when 0 is set (which is a viable option). This can be solved with array_filter($array, 'strlen')